document and release groups and teams (#4238)
alcionai/corso: CHANGELOG.md

## [Unreleased] (beta)

### Added
- Groups and Teams service support!  Channel messages and Files are now available for backup and restore in the CLI: `corso backup create groups --group '*'`
  * The cli commands for "groups" and "teams" can be used interchangably, and will operate on the same backup data.

>   * Restore and Export is available for channel files.  Channel messages are available for Export, but not Restore.

Reviewer: 2 caveats:
- Export for files is still pending
- As of now only files from public channels are backed up (ie root site)

Reviewer 2: These caveats are ok but we should do the following:
1) `Groups and Teams service support!` -> Call it Preview. CLI commands should provide the preview message also.
2) Remove Export.
3) Add the caveats to "Known issues" and limitations. The limitations being: Groups support is in preview and not complete. Files are only backed up from public channels. Export is not yet availbable.

Author: > Remove Export
Sharepoint export, you mean?  Or both sharepoint and messages?

## [Unreleased] (beta)

### Added

> - Groups and Teams service support!  Channel messages and Files are now available for backup and restore in the CLI: `corso backup create groups --group '*'`

Reviewer 2: ```suggestion
- Groups and Teams service support available as feature preview!  Channel messages and Files are now available for backup and restore in the CLI: `corso backup create groups --group '*'`
```

  * The cli commands for "groups" and "teams" can be used interchangably, and will operate on the same backup data.

>   * Channel messages are available for Export, but not Restore.
>   * Groups' SharePoint files are available for Restore, but not Export.

Reviewer 2: Remove from here - called out in limitations
```suggestion
```

  * New permissions are required to backup Channel messages.  See the [Corso Documentation](https://corsobackup.io/docs/setup/m365-access/#configure-required-permissions) for complete details.
  Even though Channel message restoration is not available, message write permissions are included to cover future integration.

>   * Support is available in an early-access status, and may be subject to breaking changes.

Reviewer 2: ```suggestion
  * This is a feature preview, and may be subject to breaking changes based on feedback and testing.
```
